src/popover.ts

```diff
@@ -453,9 +595,31 @@ export class HoverEditor extends HoverPopover {
     this.hide();
   }
 
+  shouldShow() {
+    return this.shouldShowSelf() || this.shouldShowChild();
+  }
+
+  shouldShowChild(): boolean {
+    return HoverEditor.activePopovers().some(popover => {
+      if (popover !== this && popover.targetEl && this.hoverEl.contains(popover.targetEl)) {
+        return popover.shouldShow();
+      }
+      return false;
+    });
+  }
+
   shouldShowSelf() {
     // Don't let obsidian show() us if we've already started closing
-    return !this.detaching && (this.onTarget || this.onHover);
+    // return !this.detaching && (this.onTarget || this.onHover);
+    return (
+      !this.detaching &&
+      !!(
+        this.onTarget ||
+        this.onHover ||
+        document.querySelector("body>.modal-container") ||
+        document.querySelector("body>.menu")
```

Comment:
This logic doesn't properly handle autohiding when a menu is present in the document, since menus being open doesn't prevent you from opening more popovers that then hang around.  I tried with both Pane Relief history menus and QE, and found that while hovering on the menu isn't really a problem, if you hover anything *else* you end up with a bunch of popovers that don't close themselves, and often you end up with empty popovers of "no file" because they tried to close while initializing but believed they should be shown.  The 2nd version of the menu logic from our earlier discussion doesn't have this problem.